src/main/anovos/data_ingest/data_ingest.py

```diff
@@ -41,13 +57,188 @@ def read_dataset(spark, file_path, file_type, file_configs={}):
+    treatment
+        This boolean flag provides an option to whether treat the dataframe schema or not. It might not
+        work on parquet filetype, as parquet has its own strict Schema.
+        (Default value = False)
+    id_cols
+        This argument is a list contains all the identifier columns, such as ID, Number, Code, etc
+        when treatment is set to True.
+        (Default value = [])
+    threshold_num
+        This argument determines the numerical threshold of cardinality for every column. If the number of
+        unique values of a column is larger than threshold_num, its DataType will be treated as Double.
+        Otherwise its DataType will be treated as String. (Default value = 50)
+    threshold_ratio
+        This argument determines the cardinality ratio for every column. The cardinality ratio is defined
+        by number of non-null unique values divided by number of non-null total values in a column.
+        If the cardinality ratio is larger than threshold_ratio, its DataType will be treated as Double.
+        Otherwise its DataType will be treated as String. (Default value = 0.005)
+    threshold_string
+        This argument determines the string ratio for every column. The string ratio is defined
+        by number of non-null string values divided by number of non-null total values in a column.
+        If the string ratio is smaller than threshold_ratio, its DataType will be treated as Double.
+        Otherwise its DataType will be treated as String. (Default value = 0.5)
 
     Returns
     -------
     DataFrame
 
     """
     odf = spark.read.format(file_type).options(**file_configs).load(file_path)
+    if not isinstance(parquet_conversion, bool):
+        raise TypeError("parquet_conversion must be boolean")
+    if parquet_conversion:
+        if not parquet_output_path:
+            raise ValueError(
+                "parquet_output_path cannot be blank for parquet conversion"
+            )
+    if not isinstance(treatment, bool):
+        raise TypeError("treatment must be boolean")
+    if treatment:
+        if id_cols:
+            for col in id_cols:
+                if col not in odf.columns:
+                    raise ValueError("Column not in the input dataframe: " + col)
+        if not isinstance(threshold_num, int):
+            raise TypeError("threshold_num must be integer")
+        if not isinstance(threshold_ratio, float):
+            raise TypeError("threshold_ratio must be float")
+        if threshold_ratio < 0 or threshold_ratio > 1:
+            raise ValueError("threshold_ratio value must be between 0 and 1")
+        if not isinstance(threshold_string, float):
+            raise TypeError("threshold_string must be float")
+        if threshold_string < 0 or threshold_string > 1:
+            raise ValueError("threshold_string value must be between 0 and 1")
+        if not isinstance(persist_use, bool):
+            raise TypeError("persist_use must be boolean")
```

Comment:
This kind of type checking is unnecessary and just makes the code longer and harder to read. If you add type hints, any modern IDE will alert users that they're using the wrong types. Internally you can add a tool like `mypy` to check that Anovos internally passes the proper types.

Reply:
We maintain this kind of format as of now for all Anovos functions. I think it would be wise to keep it consistent with other functions, and then we can make this a suggestion/issue and create a separated PR to fix those.

Reply:
I think it's an anti-pattern to "type-check" in this way. For example, checking for `bool` restricts the users and takes away a core feature of Python. There is nothing wrong with the following:

```python
some_list = []

read_dataset(..., persist_use=some_list, ...)

if persist_use:  # will evaluate to "false" if list is empty, otherwise "true"
    ...
```

But `isinstance(some_list, bool)` is `False`. You can do the same with most built-in types in Python like `str`, `int` etc. and there are many cases where this is fine and very convenient.

Same goes for passing an `int` instead of a `float`. There are very few cases where this is problematic, but if you have a check like `isinstance(some_integer, float)` in place you force the user to tediously do a "type-conversion" themselves, instead of having Python handle this automatically as it was intentionally designed to do.